### PR TITLE
Adds client and server applications

### PR DESCRIPTION
**Pull request by tcsc**
This patch wraps the `TaskManager` in a server application that exposes
it to the network, and adds a client application for users to interact
with the server.

Instructions for building and use are included in the `README`.

Also adds some basic system tests to test interaction between the two
parts of the system (run `make system-tests` to exercise).

### CODE REVIEW

File: cmd/levity/logs.go [hunk 1]
```diff
@@ -1 +1,40 @@
...
+	}
+)
+
+func fetchLogs(cmd *cobra.Command, args []string) {
+	request := &api.FetchLogsRequest{TaskId: &api.TaskHandle{Id: args[0]}}
+
+	client, err := makeClient()
+	if err != nil {
+		log.Fatalf("Failed to create GRPC client: %v", err)
+	}
```
Comment (russjones) [inline cmd/levity/logs.go:26]: Might want to do a a defer close.

**Review by russjones — APPROVED** [review]

File: cmd/levityd/main.go [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+	"github.com/spf13/cobra"
+	"github.com/tcsc/levity/api"
+	"github.com/tcsc/levity/taskmanager"
+	"google.golang.org/grpc"
+)
+
+var rootCmd = cobra.Command{
+	Use:   "levityd [address]",
+	Short: "Levity: A simple task mnager service",
```
Comment (r0mant) [inline cmd/levityd/main.go]: manager

File: cmd/levityd/main.go [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+	// NB: Some system tests look for some output on stdout to
+	// decide that the server is up and ready to receive requests.
+	// This interlock method is pretty flakey, and I'd definitely be
+	// looking to replace it in a live system
+
+	log.Printf("Listening on %s", listener.Addr().String())
+
+	taskMan := taskmanager.New()
+
+	grpcServer := grpc.NewServer()
```
Comment (r0mant) [inline cmd/levityd/main.go:44]: Hm, I'm not sure, is this listening without TLS, or is that still coming?

**Review by r0mant — APPROVED** [review]
lgtm provided the security bits are coming in the next PR

**Review by fspmarshall — APPROVED** [review]

File: cmd/levityd/main.go (hunk 3 above)
Comment (tcsc) [reply cmd/levityd/main.go:44]: TLS coming in the next patch

**Pull request merged**